merkledag/README.md

```diff
@@ -0,0 +1,144 @@
+# The merkledag
+
+
+Authors: [Juan Benet](github.com/jbenet)
+
+Reviewers:
+
+- [Jeromy Johnson](github.com/whyrusleeping)
+
+* * *
+
+This is the [Spec](../) for the IPFS Merkledag.
+
+## Definitions
+
+- `hash` - throughout this document, the word `hash` refers specifically to cryptographic hash functions, such as sha3.
+- `dag` - directed acyclic graph
+- `merkle-link` - a link (graph edge) between two objects, which is (a) represented by the hash of the target object, and (b) embedded in the source object. merkle-links construct graphs (dags) whose links are content-addressed, and authenticated.
+- `merkledag` - the merkledag is a directed acyclic graph whose links are merkle-links (hashes of the content). It is a hash tree, and (under a very lose definition) a merkle tree. Alternative names: the merkle-web, the merkle-forest, the merkle-chain.
+- `multihash` - the [multihash](https://github.com/jbenet/multihash) format / protocol.
+- `ipfs object` - a node in the ipfs merkledag. It represents a singular entity.
+- `merkledag format` - the format that ipfs objects are expressed with.
+- `link segment` or `link table` - the part of the merkledag format that expresses links to other objects.
+- `data segment` - the part of the merkledag format that expresses object data.
+- `protobuf` - protocol buffers, a serialization encoding.
+- `multicodec` - a self-describing, generalized serialization format.
```

Comment:
Is there an existing spec for this?  Searching around for “multicodec” didn't turn up any obvious hits.